dens-transactions/src/DensTransactions/DensTransactions.ts

```diff
@@ -1,3 +1,229 @@
+
+  const complete = await tx.complete();
+  const signed = complete.sign();
+
+  const readyToSubmit = await signed.complete();
+  const hash = await readyToSubmit.submit();
+
+  await new Promise((r) => setTimeout(r, 10000));
+  const walletUTXOs = await lucid.wallet.getUtxos();
+
+  const utxoWithHash = walletUTXOs.find((x) => x.txHash === hash);
+
+  return { txHash: hash, outputIndex: utxoWithHash.outputIndex };
+};
+
+export const initializeDeNS = async (
+  lucid: L.Lucid,
+  params: Utils.DeNSParams,
+  _path: string,
+  oneShotOutRef: L.OutRef,
+): Promise<L.Tx> => {
+  const builder = new L.Tx(lucid);
+  const utils = new L.Utils(lucid);
+  const initialSetDatumPD = IsPlutusData[SetDatum].toData(
+    Utils.initialSetDatum,
+  );
+  logger.info(
+    "Initial set datum: " + JSON.stringify(initialSetDatumPD, null, 4),
+  );
+  const initialSetDatumCSL = (Utils.toCslPlutusData(initialSetDatumPD))
+    .to_hex();
+  const initialSetDatumDatum: L.OutputData = { inline: initialSetDatumCSL };
+  const protocolDatumRaw: Protocol = {
+    elementIdMintingPolicy: Utils.mkLBScriptHash(params.elemIDPolicy),
+    setElemMintingPolicy: Utils.mkLBScriptHash(params.setElemIDPolicy),
+    setValidator: Utils.mkLBScriptHash(params.setValidator),
+    recordsValidator: Utils.mkLBScriptHash(params.recordValidator),
+  };
+
+  const protocolDatum: L.OutputData = {
+    inline: Utils.toCslPlutusData(
+      IsPlutusData[Protocol].toData(protocolDatumRaw),
+    )
+      .to_hex(),
+  };
+
+  const oneShotUtxos = await lucid.utxosByOutRef([oneShotOutRef]);
+  const oneShotUtxo = oneShotUtxos[0];
+
+  // Mint one protocol token
+  const protocolPolicyID = utils.mintingPolicyToId(params.protocolPolicy);
+
+  const oneProtocolToken = { [protocolPolicyID]: BigInt(1) };
+
+  // Mint one setElem token
+  const setElemPolicyID = utils.mintingPolicyToId(params.setElemIDPolicy);
+
+  const oneSetElemToken = { [setElemPolicyID]: BigInt(1) };
+
+  const setValidatorAddr = utils.validatorToAddress(params.setValidator);
+
+  // TODO: Figure out how to make a unit datum
+  return builder
+    .attachMintingPolicy(params.protocolPolicy)
+    .mintAssets(oneProtocolToken, L.Data.void())
+    .attachMintingPolicy(params.setElemIDPolicy)
+    .mintAssets(oneSetElemToken, initSetInsert().to_hex())
+    // WARNING(jaredponn): we MUST put the protocol datum as a tx output BEFORE
+    // ALL OTHER TX OUTPUTS. This is because of the way the query layer works.
+    // There is a way to "fix" this, but that'd make a good separate project --
+    // A Framework for Efficient Databases of subsets of UTxOs for Cardano.
+    .payToAddressWithData(setValidatorAddr, protocolDatum, oneProtocolToken)
+    .payToAddressWithData(
+      setValidatorAddr,
+      initialSetDatumDatum,
+      oneSetElemToken,
+    )
+    .collectFrom([oneShotUtxo]);
+};
+
+export const registerDomain = async (
+  lucid: L.Lucid,
+  params: Utils.DeNSParams,
+  domain: string,
+  path: string,
```

Comment:
Would it be possible to for this to support both unix domain sockets and the normal IP mechanisms? We can case on the type from 
```
import { UnixDomainOrInternetDomain } from "lbf-dens-db/LambdaBuffers/Dens/Config.mjs"
```